plugin/option/boot_files.go

```diff
@@ -0,0 +1,52 @@
+package option
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/iana"
+	"github.com/nextdhcp/nextdhcp/core/option"
+)
+
+/*** Client Arch type
+	0    Intel x86PC
+	1    NEC/PC98
+	2    EFI Itanium
+	3    DEC Alpha
+	4    Arc x86
+	5    Intel Lean Client
+	6    EFI IA32
+	7    EFI BC
+	8    EFI Xscale
+	9    EFI x86-64
+***/
+
+const (
+	// TODO: will set in database and get from database
+	efiBootFile    = "ipxe.efi"
+	legacyBootFile = "undionly.kpxe"
+)
+
+func (p *Plugin) GetBootFileOpt(ctx context.Context, req, res *dhcpv4.DHCPv4) (*dhcpv4.Option, error) {
+	bootFileName := p.parseBootFileName(req)
+	code, value, err := option.ParseKnown("filename", []string{bootFileName})
+	if err != nil {
+		return nil, err
+	}
+	option := dhcpv4.OptGeneric(code, value.ToBytes())
+	return &option, nil
+}
+
+func (p *Plugin) parseBootFileName(req *dhcpv4.DHCPv4) string {
+	archs := iana.Archs(req.ClientArch())
+
+	rex := regexp.MustCompile("(^EFI \\S+$)")
+	bootFile := legacyBootFile
```

Comment:
I'd suggestion doing that on package level once at initialization time so we don't need to re-compile the RE for each DHCP request.